File: src/Kentico.Community.Portal.Core/Content/UTMParametersDataType.cs
namespace Kentico.Community.Portal.Core.Content;

public class UTMParametersDataType
{
    public const string FIELD_TYPE = "utmparameters";

    public Guid ID { get; set; } = Guid.NewGuid();
    public string Source { get; set; } = "";
    public string Medium { get; set; } = "";
    public string Campaign { get; set; } = "";
    public string Content { get; set; } = "";
    public string Term { get; set; } = "";
}

File: src/Kentico.Community.Portal.Admin/Client/src/features/utm-parameters/UTMParametersDataType.ts
export type UTMParametersDataType = {
  id: string;
  source: string;
  medium: string;
  campaign: string;
  content: string;
  term: string;
};

export type UTMParameterDataTypeField = Exclude<
  keyof UTMParametersDataType,
  'id'
>;

export function newUTMParameters(): UTMParametersDataType {
  return {
    id: crypto.randomUUID(),
    source: '',
    medium: '',
    campaign: '',
    content: '',
    term: '',
  };
}
